rkt/image/httpops.go

```diff
@@ -91,19 +91,19 @@ func (o *httpOps) DownloadImageWithETag(u *url.URL, etag string) (readSeekCloser
 	if err != nil {
 		return nil, nil, err
 	}
-	defer func() { maybeClose(aciFile) }()
 
 	session := o.getSession(u, aciFile.File, "ACI", etag)
 	dl := o.getDownloader(session)
 	if err := dl.Download(u, aciFile.File); err != nil {
+		aciFile.Close() // download failed, close the aciFile to implicitely release the lock
```

Comment:
And also agreed, that this PR introduces a stylistic nuance. Nevertheless I believe that (fortunately, or unfortunately) this is Go's nature, namely being pretty verbose, especially in error cases. The biggest pro of `maybeClose` is, that it reduces LOC, but it made it hard to reason about when these files were actually "really" closed. This will make the reasoning easier at the expense of verbosity.